### ISSUE TITLE
Results Page: Allow users to change the page title of PerfCompare and have the title persist in the url

### ISSUE THREAD

**Issue by Carla-Moz** [mozilla/perfcompare#769]
**Description:** As a user, I want to be able to change the title of PerfCompare and have the new title persist in the url, so that I can share the updated title with others.

**Steps to Reproduce:**

1. Read the PerfCompare Documentation to get familiar with the app.
2. Navigate to the title section of the PerfCompare results: [ResultsPage](https://main--mozilla-perfcompare.netlify.app/compare-results?baseRev=901eda21e48867cdcc016a0dba81e51c0d4dd16b&baseRepo=mozilla-central&newRev=b48e31d47d1f562424c7693ad93f74ed39251edb&newRepo=mozilla-central&newRev=e5eef47a4db1acee5e17f5b4ea6219301d995c63&newRepo=mozilla-central&framework=13.)
3. Observe that there is currently no option to modify the title and have it persist in the URL.

![image](https://github.com/user-attachments/assets/ce4e889c-c9ab-413d-9420-0b642321fefd)

**Acceptance Criteria:**
1. The title should be editable by the user.
2. The new title should persist in the URL so that it is retained after a page refresh.
3. When a user shares the URL with others, the custom title should be displayed.
4. The url should have the updated page title.

**Comment by FatumaA** [comment]
Hi,
I'd like to attempt this

**Comment by Carla-Moz** [comment]
@FatumaA, Please go ahead and try.

**Comment by syedbarimanjan** [comment]
@Carla-Moz what do you mean by the title?

**Comment by hibaa03** [comment]
Hi! I've implemented functionality when I enter a custom title in the input field, it updates the URL correctly and when i refresh the page, it persists.
If I share the URL, the custom title is displayed as intended and the title is in the URL as well, as for me locally it shows (http://localhost:3000/compare-results?baseRev=aa73934529cc8e33ed70fd861d70456fdf7f0818&baseRepo=try&framework=1&**title=updated+title**) 

However, when navigating back to the Results page from the home page, the previously entered title is not retained, and the title reverts to a default value.

My question is, Is this behavior expected, or should the custom title persist across navigations? If it's expected to retain the custom title, what changes would be necessary?

**Comment by syedbarimanjan** [comment]
> Hi! I've implemented functionality when I enter a custom title in the input field, it updates the URL correctly and when i refresh the page, it persists. If I share the URL, the custom title is displayed as intended and the title is in the URL as well, as for me locally it shows (http://localhost:3000/compare-results?baseRev=aa73934529cc8e33ed70fd861d70456fdf7f0818&baseRepo=try&framework=1&**title=updated+title**)
> 
> However, when navigating back to the Results page from the home page, the previously entered title is not retained, and the title reverts to a default value.
> 
> My question is, Is this behavior expected, or should the custom title persist across navigations? If it's expected to retain the custom title, what changes would be necessary?

Yes i think so that this is the expected behavior as stated [here](https://github.com/mozilla/perfcompare/pull/724#discussion_r1723672497).

**Comment by FatumaA** [comment]
Hi Carla, 
I've linked my first attempt at this. Please let me know if that was what was meant and looking forward to you review

**Comment by claire256** [comment]
Hello, @Carla-Moz ,What title do you mean?

**Comment by hibaa03** [comment]
@claire256 from what I know, this is what page the title is ![image](https://github.com/user-attachments/assets/354fbeea-4e03-4878-8ba8-ccc35cf20772)

**Comment by claire256** [comment]
@hibaa03 , thanks.
